src/core/test/system/borrower/balance.t.sol

```diff
@@ -0,0 +1,131 @@
+        uint takeAmount = currency.balanceOf(address(junior));
+        borrower.balance();
+        assertPostConditionTake(takeAmount, initialCurrencyBalance);
+    }
+
+    function balanceGive() public {
+        uint initialCurrencyBalance = currency.balanceOf(address(shelf));
+        uint giveAmount = currency.balanceOf(address(junior));
+        borrower.balance();
+        assertPostConditionGive(giveAmount, initialCurrencyBalance);
+    }
+
+    function assertPreConditionTake(uint takeAmount) public {
+        // assert: borrowFromTranches is active
+        assert(distributor.borrowFromTranches());
+        // assert: junior has enough capital
+        assert(currency.balanceOf(address(junior)) == takeAmount);
+    }
+
+    function assertPostConditionTake(uint takeAmount, uint initialCurrencyBalance) public {
+        // assert: takeAmount > 0
+        assert(takeAmount > 0);
+        // assert: all funds transferred from tranche reserve
+        assertEq(currency.balanceOf(address(junior)), 0);
+        // assert: junior received funds
+        assertEq(currency.balanceOf(address(shelf)), add(initialCurrencyBalance, takeAmount));
+    }
+
+    function assertPreConditionGive(uint giveAmount) public {
+        // assert: borrowFromTranches is inactive
+        assert(!distributor.borrowFromTranches());
+        // assert: shelf has funds
+        assert(currency.balanceOf(address(shelf)) == giveAmount);
+    }
+
+    function assertPostConditionGive(uint giveAmount, uint initialJuniorBalance) public {
+        // assert: giveAmount > 0
+        assert(giveAmount > 0);
+        // assert: all funds transferred from shelf
+        assertEq(currency.balanceOf(address(shelf)), 0);
+        // assert: all funds transferred from shelf
+        assertEq(currency.balanceOf(address(junior)), add(initialJuniorBalance, giveAmount));
+    }
+
+    function testBalanceTake() public {
+        uint takeAmount = 100 ether;
+        // supply junior tranche with funds
+        supplyFunds(takeAmount, address(junior));
+        assertPreConditionTake(takeAmount);
+        // take money from tranche and transfer into shelf
+        balanceTake();
+    }
+
+    function testFailBalanceTakeNoFundsAvailable() public {
+        // junior tranche not supplied with capital
+        balanceTake();
+    }
+
+    function testFailBalanceTakeInactive() public {
+        uint takeAmount = 100 ether;
+        // supply junior tranche with funds
+        supplyFunds(takeAmount, address(junior));
+        distributor.file("borrowFromTranches", false);
+        balanceTake();
+    }
+
+    function testBalanceGive() public {
+        uint giveAmount = 100 ether;
+        // supply shelf tranche with funds
+        supplyFunds(giveAmount, address(junior));
+        // deactivate borrow from tranches
+        distributor.file("borrowFromTranches", false);
+        // take money from shelf and transfer into shelf
```

Comment:
shouldn't this test be testing repay? then the shelf should have money to transfer to junior

Reply:
good catch. I supplied the opposite side with funds